src/components/SearchBar.tsx

```diff
@@ -16,11 +16,32 @@ export interface SearchBarState {
     detailed: boolean;
 }
 
+interface SearchDisplayModeChangedFunc {
+    (value: boolean): void;
+}
```

Comment:
Thanks @yeexinc! One thing though, from looking at this function prototype I am not sure what `value === true` means. Does it mean `structured` or `list` display? This is the reason why `onStructuredModeChanged` and `onDetailedModeChanged` event handlers cannot share the same `SearchDisplayModeChangedFunc`. We need different ones for each of them, and also more meaningful names than `value: boolean` (if it was `structured: boolean`, it tells me that `true` means structured view).